From 43f3829d651ca9a0367252bbfe0c83260b892990 Mon Sep 17 00:00:00 2001
From: Suneel Garapati <sgarapati@caviumnetworks.com>
Date: Mon, 23 Sep 2019 13:02:29 -0700
Subject: [PATCH 0963/1239] octeontx: octeontx2: config cleanup

- cleanup configs with savedefconfig
- remove redundant and simplify config macros
- merge 81xx and 83xx config headers

Change-Id: I64bac1ec587a77b87ad91c44d7154971f6dbe189
Signed-off-by: Suneel Garapati <sgarapati@caviumnetworks.com>
Reviewed-on: https://sj1git1.cavium.com/16341
---
 arch/arm/include/asm/arch-octeontx/octeontx.h |  14 ++
 arch/arm/mach-octeontx/Kconfig                |  18 ++-
 board/Marvell/octeontx/Kconfig                |  20 ++-
 board/Marvell/octeontx/fdt.c                  |  12 +-
 board/Marvell/octeontx/octeontx.c             |   2 +-
 board/Marvell/octeontx2/octeontx2.c           |   2 +-
 cmd/nvedit.c                                  |  14 --
 configs/octeontx2_95xx_defconfig              |  29 ++--
 configs/octeontx2_96xx_defconfig              |  29 ++--
 configs/octeontx2_98xx_defconfig              |  24 ++--
 configs/octeontx2_loki_defconfig              |  25 ++--
 configs/octeontx_81xx_defconfig               |  20 +--
 configs/octeontx_83xx_defconfig               |  18 +--
 drivers/mmc/Kconfig                           |  35 ++---
 drivers/mmc/octeontx_hsmmc.c                  |  12 +-
 drivers/net/octeontx/nic_main.c               |   7 +-
 drivers/net/octeontx/octeontx_bgx.c           |  30 ++---
 drivers/net/octeontx/octeontx_bgx.h           |   4 +-
 include/configs/octeontx2_95xx.h              |  22 +--
 include/configs/octeontx2_96xx.h              |  23 +---
 include/configs/octeontx2_98xx.h              |  23 +---
 include/configs/octeontx2_loki.h              |  20 +--
 include/configs/octeontx_83xx.h               | 127 ------------------
 .../{octeontx_81xx.h => octeontx_common.h}    |  36 ++---
 24 files changed, 172 insertions(+), 394 deletions(-)
 delete mode 100644 include/configs/octeontx_83xx.h
 rename include/configs/{octeontx_81xx.h => octeontx_common.h} (78%)

diff --git a/arch/arm/include/asm/arch-octeontx/octeontx.h b/arch/arm/include/asm/arch-octeontx/octeontx.h
index 7aa08afbb3..ac2bc86a30 100644
--- a/arch/arm/include/asm/arch-octeontx/octeontx.h
+++ b/arch/arm/include/asm/arch-octeontx/octeontx.h
@@ -13,6 +13,20 @@
 #define is_board_model(model)  (g_cavm_bdt.prod_id == (model))
 
 #define MAX_LMAC_PER_BGX 4
+#define LMAC_CNT MAX_LMAC_PER_BGX
+
+#if defined(CONFIG_TARGET_OCTEONTX_81XX)
+
+/** Maximum number of BGX interfaces per CPU node */
+#define MAX_BGX_PER_NODE	3
+#define OCTEONTX_XCV	/* RGMII Interface */
+
+#elif defined(CONFIG_TARGET_OCTEONTX_83XX)
+
+/** Maximum number of BGX interfaces per CPU node */
+#define MAX_BGX_PER_NODE	4
+
+#endif
 
 /** Reg offsets */
 #define CAVM_MIO_FUS_DAT2	0x87E003001410ULL
diff --git a/arch/arm/mach-octeontx/Kconfig b/arch/arm/mach-octeontx/Kconfig
index cb520f5ed9..28ecf9821f 100644
--- a/arch/arm/mach-octeontx/Kconfig
+++ b/arch/arm/mach-octeontx/Kconfig
@@ -1,14 +1,18 @@
 if ARCH_OCTEONTX
 
-config SYS_VENDOR
-	string
-	default	"Marvell"
+choice
+	prompt "OcteonTX board select"
+	optional
 
-config SYS_SOC
-	string
-	default "octeontx"
+config TARGET_OCTEONTX_81XX
+	bool "Marvell OcteonTX CN81XX"
 
-config SYS_BOARD
+config TARGET_OCTEONTX_83XX
+	bool "Marvell OcteonTX CN83XX"
+
+endchoice
+
+config SYS_SOC
 	string
 	default "octeontx"
 
diff --git a/board/Marvell/octeontx/Kconfig b/board/Marvell/octeontx/Kconfig
index 016d024aef..45d115916c 100644
--- a/board/Marvell/octeontx/Kconfig
+++ b/board/Marvell/octeontx/Kconfig
@@ -1,18 +1,14 @@
-if ARCH_OCTEONTX
+if TARGET_OCTEONTX_81XX || TARGET_OCTEONTX_83XX
 
-choice
-	prompt "Target board"
+config SYS_VENDOR
+	string
+	default	"Marvell"
 
-config	TARGET_OCTEONTX_83XX
-	bool "OcteonTX CN83XX"
-
-config	TARGET_OCTEONTX_81XX
-	bool "OcteonTX CN81XX"
-
-endchoice
+config SYS_BOARD
+	string
+	default "octeontx"
 
 config SYS_CONFIG_NAME
-	default "octeontx_81xx" if TARGET_OCTEONTX_81XX
-	default "octeontx_83xx" if TARGET_OCTEONTX_83XX
+	default "octeontx_common"
 
 endif
diff --git a/board/Marvell/octeontx/fdt.c b/board/Marvell/octeontx/fdt.c
index 7b2b2bfdd5..2a43cfdb66 100644
--- a/board/Marvell/octeontx/fdt.c
+++ b/board/Marvell/octeontx/fdt.c
@@ -75,12 +75,12 @@ void octeontx_parse_phy_info(void)
 
 	offset = fdt_node_offset_by_compatible(fdt, -1, "pci-bridge");
 	if (offset > 1) {
-		for (bgx_id = 0; bgx_id < CONFIG_MAX_BGX_PER_NODE; bgx_id++) {
-			int phy_addr[MAX_LMAC_PER_BGX] = { [0 ... MAX_LMAC_PER_BGX - 1] = -1};
-			bool autoneg_dis[MAX_LMAC_PER_BGX] = { [0 ... MAX_LMAC_PER_BGX - 1] = 0};
-			int mdio_bus[MAX_LMAC_PER_BGX] = { [0 ... MAX_LMAC_PER_BGX - 1] = -1};
-			bool lmac_reg[MAX_LMAC_PER_BGX] = { [0 ... MAX_LMAC_PER_BGX - 1] = 0};
-			bool lmac_enable[MAX_LMAC_PER_BGX] = { [0 ... MAX_LMAC_PER_BGX - 1] = 0};
+		for (bgx_id = 0; bgx_id < MAX_BGX_PER_NODE; bgx_id++) {
+			int phy_addr[LMAC_CNT] = {[0 ... LMAC_CNT - 1] = -1};
+			bool autoneg_dis[LMAC_CNT] = {[0 ... LMAC_CNT - 1] = 0};
+			int mdio_bus[LMAC_CNT] = {[0 ... LMAC_CNT - 1] = -1};
+			bool lmac_reg[LMAC_CNT] = {[0 ... LMAC_CNT - 1] = 0};
+			bool lmac_enable[LMAC_CNT] = {[0 ... LMAC_CNT - 1] = 0};
 
 			snprintf(bgxname, sizeof(bgxname),
 				 "bgx%d", bgx_id);
diff --git a/board/Marvell/octeontx/octeontx.c b/board/Marvell/octeontx/octeontx.c
index 748965c26f..7a24134f88 100644
--- a/board/Marvell/octeontx/octeontx.c
+++ b/board/Marvell/octeontx/octeontx.c
@@ -31,7 +31,7 @@ void octeontx_cleanup_ethaddr(void)
 	for (int i = 0; i < 20; i++) {
 		sprintf(ename, i ? "eth%daddr" : "ethaddr", i);
 		if (env_get(ename))
-			env_set_force(ename, NULL);
+			env_set(ename, NULL);
 	}
 }
 
diff --git a/board/Marvell/octeontx2/octeontx2.c b/board/Marvell/octeontx2/octeontx2.c
index 1a7956f1c2..5de4e35bf0 100644
--- a/board/Marvell/octeontx2/octeontx2.c
+++ b/board/Marvell/octeontx2/octeontx2.c
@@ -37,7 +37,7 @@ void octeontx2_cleanup_ethaddr(void)
 	for (int i = 0; i < 20; i++) {
 		sprintf(ename, i ? "eth%daddr" : "ethaddr", i);
 		if (env_get(ename))
-			env_set_force(ename, NULL);
+			env_set(ename, NULL);
 	}
 }
 
diff --git a/cmd/nvedit.c b/cmd/nvedit.c
index 55221ac53f..1cb0bc1460 100644
--- a/cmd/nvedit.c
+++ b/cmd/nvedit.c
@@ -313,20 +313,6 @@ int env_set(const char *varname, const char *varvalue)
 		return _do_env_set(0, 3, (char * const *)argv, H_PROGRAMMATIC);
 }
 
-int env_set_force(const char *varname, const char *varvalue)
-{
-	const char * const argv[5] = { "setenv", "-f", varname, varvalue, NULL };
-
-	/* before import into hashtable */
-	if (!(gd->flags & GD_FLG_ENV_READY))
-		return 1;
-
-	if (varvalue == NULL || varvalue[0] == '\0')
-		return _do_env_set(0, 3, (char * const *)argv, H_FORCE);
-	else
-		return _do_env_set(0, 4, (char * const *)argv, H_FORCE);
-}
-
 /**
  * Set an environment variable to an integer value
  *
diff --git a/configs/octeontx2_95xx_defconfig b/configs/octeontx2_95xx_defconfig
index c63d7278d3..4bb03ef67a 100644
--- a/configs/octeontx2_95xx_defconfig
+++ b/configs/octeontx2_95xx_defconfig
@@ -16,8 +16,12 @@ CONFIG_BOARD_EARLY_INIT_R=y
 CONFIG_HUSH_PARSER=y
 CONFIG_SYS_PROMPT="Marvell> "
 # CONFIG_CMD_BOOTEFI_HELLO_COMPILE is not set
+CONFIG_CMD_MD5SUM=y
+CONFIG_MD5SUM_VERIFY=y
 CONFIG_CMD_MEMTEST=y
 CONFIG_CMD_MX_CYCLIC=y
+CONFIG_CMD_SHA1SUM=y
+CONFIG_SHA1SUM_VERIFY=y
 # CONFIG_CMD_FLASH is not set
 CONFIG_CMD_GPIO=y
 CONFIG_CMD_I2C=y
@@ -37,6 +41,8 @@ CONFIG_CMD_SNTP=y
 CONFIG_CMD_DNS=y
 CONFIG_CMD_LINK_LOCAL=y
 CONFIG_CMD_BOOTIMGUP=y
+CONFIG_CMD_TIME=y
+CONFIG_CMD_FAILSAFE=y
 CONFIG_CMD_ETH=y
 CONFIG_CMD_EXT2=y
 CONFIG_CMD_EXT4=y
@@ -53,17 +59,9 @@ CONFIG_DM_GPIO=y
 CONFIG_I2C_GPIO=y
 CONFIG_DM_I2C=y
 CONFIG_MISC=y
-CONFIG_MMC=y
-CONFIG_MMC_HS200_SUPPORT=y
-CONFIG_MMC_HS400_SUPPORT=y
-CONFIG_MMC_WRITE=y
-CONFIG_MMC_SUPPORTS_TUNING=y
 CONFIG_DM_MMC=y
-CONFIG_DM_REGULATOR=y
-CONFIG_DM_REGULATOR_FIXED=y
-CONFIG_DM_REGULATOR_GPIO=y
-CONFIG_DM_RTC=y
-CONFIG_RTC_DS1337=y
+CONFIG_MMC_HS400_SUPPORT=y
+CONFIG_MMC_OCTEONTX=y
 CONFIG_MTD=y
 CONFIG_DM_SPI_FLASH=y
 CONFIG_SPI_FLASH=y
@@ -78,6 +76,11 @@ CONFIG_PCI=y
 CONFIG_DM_PCI=y
 CONFIG_DM_PCI_COMPAT=y
 CONFIG_PCI_OCTEONTX_ECAM=y
+CONFIG_DM_REGULATOR=y
+CONFIG_DM_REGULATOR_FIXED=y
+CONFIG_DM_REGULATOR_GPIO=y
+CONFIG_DM_RTC=y
+CONFIG_RTC_DS1337=y
 CONFIG_DM_SERIAL=y
 CONFIG_DEBUG_UART_PL011=y
 CONFIG_DEBUG_UART_BASE=0x87e028000000
@@ -88,9 +91,3 @@ CONFIG_DM_SPI=y
 CONFIG_OCTEONTX_SPI=y
 CONFIG_FAT_WRITE=y
 CONFIG_ERRNO_STR=y
-CONFIG_CMD_SHA1SUM=y
-CONFIG_SHA1SUM_VERIFY=y
-CONFIG_CMD_MD5SUM=y
-CONFIG_MD5SUM_VERIFY=y
-CONFIG_CMD_TIME=y
-CONFIG_CMD_FAILSAFE=y
diff --git a/configs/octeontx2_96xx_defconfig b/configs/octeontx2_96xx_defconfig
index b09c68ac58..ac894fb32e 100644
--- a/configs/octeontx2_96xx_defconfig
+++ b/configs/octeontx2_96xx_defconfig
@@ -18,17 +18,16 @@ CONFIG_BOARD_EARLY_INIT_R=y
 CONFIG_HUSH_PARSER=y
 CONFIG_SYS_PROMPT="Marvell> "
 # CONFIG_CMD_BOOTEFI_HELLO_COMPILE is not set
+CONFIG_CMD_MD5SUM=y
+CONFIG_MD5SUM_VERIFY=y
 CONFIG_CMD_MEMTEST=y
 CONFIG_CMD_MX_CYCLIC=y
+CONFIG_CMD_SHA1SUM=y
+CONFIG_SHA1SUM_VERIFY=y
 # CONFIG_CMD_FLASH is not set
 CONFIG_CMD_GPIO=y
 CONFIG_CMD_I2C=y
 CONFIG_CMD_MMC=y
-CONFIG_MMC_HS200_SUPPORT=y
-CONFIG_MMC_HS400_SUPPORT=y
-CONFIG_MMC_WRITE=y
-CONFIG_MMC_SUPPORTS_TUNING=y
-CONFIG_MMC=y
 CONFIG_CMD_PART=y
 CONFIG_CMD_PCI=y
 CONFIG_CMD_SF=y
@@ -45,6 +44,8 @@ CONFIG_CMD_SNTP=y
 CONFIG_CMD_DNS=y
 CONFIG_CMD_LINK_LOCAL=y
 CONFIG_CMD_BOOTIMGUP=y
+CONFIG_CMD_TIME=y
+CONFIG_CMD_FAILSAFE=y
 CONFIG_CMD_ETH=y
 CONFIG_CMD_EXT2=y
 CONFIG_CMD_EXT4=y
@@ -67,11 +68,9 @@ CONFIG_I2C_SET_DEFAULT_BUS_NUM=y
 CONFIG_I2C_MUX=y
 CONFIG_I2C_MUX_PCA954x=y
 CONFIG_MISC=y
-CONFIG_MMC=y
 CONFIG_DM_MMC=y
-CONFIG_DM_REGULATOR=y
-CONFIG_DM_REGULATOR_FIXED=y
-CONFIG_DM_REGULATOR_GPIO=y
+CONFIG_MMC_HS400_SUPPORT=y
+CONFIG_MMC_OCTEONTX=y
 CONFIG_MTD=y
 CONFIG_DM_SPI_FLASH=y
 CONFIG_SPI_FLASH=y
@@ -92,6 +91,9 @@ CONFIG_PCI=y
 CONFIG_DM_PCI=y
 CONFIG_DM_PCI_COMPAT=y
 CONFIG_PCI_OCTEONTX_ECAM=y
+CONFIG_DM_REGULATOR=y
+CONFIG_DM_REGULATOR_FIXED=y
+CONFIG_DM_REGULATOR_GPIO=y
 CONFIG_DM_RTC=y
 CONFIG_RTC_DS1337=y
 CONFIG_SCSI=y
@@ -116,12 +118,3 @@ CONFIG_USB_ETHER_ASIX88179=y
 CONFIG_USB_ETHER_RTL8152=y
 CONFIG_FAT_WRITE=y
 CONFIG_ERRNO_STR=y
-CONFIG_CMD_SHA1SUM=y
-CONFIG_SHA1SUM_VERIFY=y
-CONFIG_CMD_MD5SUM=y
-CONFIG_MD5SUM_VERIFY=y
-CONFIG_CMD_TIME=y
-CONFIG_CONSOLE_MUX=y
-CONFIG_SYS_CONSOLE_IS_IN_ENV=y
-CONFIG_CMD_FAILSAFE=y
-# CONFIG_CMD_ATTEST is not set
diff --git a/configs/octeontx2_98xx_defconfig b/configs/octeontx2_98xx_defconfig
index 6bd0f05ba3..dc846748bf 100644
--- a/configs/octeontx2_98xx_defconfig
+++ b/configs/octeontx2_98xx_defconfig
@@ -18,17 +18,16 @@ CONFIG_BOARD_EARLY_INIT_R=y
 CONFIG_HUSH_PARSER=y
 CONFIG_SYS_PROMPT="Marvell> "
 # CONFIG_CMD_BOOTEFI_HELLO_COMPILE is not set
+CONFIG_CMD_MD5SUM=y
+CONFIG_MD5SUM_VERIFY=y
 CONFIG_CMD_MEMTEST=y
 CONFIG_CMD_MX_CYCLIC=y
+CONFIG_CMD_SHA1SUM=y
+CONFIG_SHA1SUM_VERIFY=y
 # CONFIG_CMD_FLASH is not set
 CONFIG_CMD_GPIO=y
 CONFIG_CMD_I2C=y
 CONFIG_CMD_MMC=y
-CONFIG_MMC_HS200_SUPPORT=y
-CONFIG_MMC_HS400_SUPPORT=y
-CONFIG_MMC_WRITE=y
-CONFIG_MMC_SUPPORTS_TUNING=y
-CONFIG_MMC=y
 CONFIG_CMD_PART=y
 CONFIG_CMD_PCI=y
 CONFIG_CMD_SF=y
@@ -45,6 +44,7 @@ CONFIG_CMD_SNTP=y
 CONFIG_CMD_DNS=y
 CONFIG_CMD_LINK_LOCAL=y
 CONFIG_CMD_BOOTIMGUP=y
+CONFIG_CMD_TIME=y
 CONFIG_CMD_EXT2=y
 CONFIG_CMD_EXT4=y
 CONFIG_CMD_EXT4_WRITE=y
@@ -66,11 +66,9 @@ CONFIG_I2C_SET_DEFAULT_BUS_NUM=y
 CONFIG_I2C_MUX=y
 CONFIG_I2C_MUX_PCA954x=y
 CONFIG_MISC=y
-CONFIG_MMC=y
 CONFIG_DM_MMC=y
-CONFIG_DM_REGULATOR=y
-CONFIG_DM_REGULATOR_FIXED=y
-CONFIG_DM_REGULATOR_GPIO=y
+CONFIG_MMC_HS400_SUPPORT=y
+CONFIG_MMC_OCTEONTX=y
 CONFIG_MTD=y
 CONFIG_DM_SPI_FLASH=y
 CONFIG_SPI_FLASH=y
@@ -91,6 +89,9 @@ CONFIG_PCI=y
 CONFIG_DM_PCI=y
 CONFIG_DM_PCI_COMPAT=y
 CONFIG_PCI_OCTEONTX_ECAM=y
+CONFIG_DM_REGULATOR=y
+CONFIG_DM_REGULATOR_FIXED=y
+CONFIG_DM_REGULATOR_GPIO=y
 CONFIG_DM_RTC=y
 CONFIG_RTC_DS1337=y
 CONFIG_SCSI=y
@@ -114,8 +115,3 @@ CONFIG_USB_ETHER_ASIX88179=y
 CONFIG_USB_ETHER_RTL8152=y
 CONFIG_FAT_WRITE=y
 CONFIG_ERRNO_STR=y
-CONFIG_CMD_SHA1SUM=y
-CONFIG_SHA1SUM_VERIFY=y
-CONFIG_CMD_MD5SUM=y
-CONFIG_MD5SUM_VERIFY=y
-CONFIG_CMD_TIME=y
diff --git a/configs/octeontx2_loki_defconfig b/configs/octeontx2_loki_defconfig
index 25e33e7d49..d403b5c44a 100644
--- a/configs/octeontx2_loki_defconfig
+++ b/configs/octeontx2_loki_defconfig
@@ -16,8 +16,12 @@ CONFIG_BOARD_EARLY_INIT_R=y
 CONFIG_HUSH_PARSER=y
 CONFIG_SYS_PROMPT="Marvell> "
 # CONFIG_CMD_BOOTEFI_HELLO_COMPILE is not set
+CONFIG_CMD_MD5SUM=y
+CONFIG_MD5SUM_VERIFY=y
 CONFIG_CMD_MEMTEST=y
 CONFIG_CMD_MX_CYCLIC=y
+CONFIG_CMD_SHA1SUM=y
+CONFIG_SHA1SUM_VERIFY=y
 # CONFIG_CMD_FLASH is not set
 CONFIG_CMD_GPIO=y
 CONFIG_CMD_I2C=y
@@ -37,6 +41,8 @@ CONFIG_CMD_SNTP=y
 CONFIG_CMD_DNS=y
 CONFIG_CMD_LINK_LOCAL=y
 CONFIG_CMD_BOOTIMGUP=y
+CONFIG_CMD_TIME=y
+CONFIG_CMD_FAILSAFE=y
 CONFIG_CMD_ETH=y
 CONFIG_CMD_EXT2=y
 CONFIG_CMD_EXT4=y
@@ -53,15 +59,9 @@ CONFIG_DM_GPIO=y
 CONFIG_I2C_GPIO=y
 CONFIG_DM_I2C=y
 CONFIG_MISC=y
-CONFIG_MMC=y
-CONFIG_MMC_HS200_SUPPORT=y
-CONFIG_MMC_HS400_SUPPORT=y
-CONFIG_MMC_WRITE=y
-CONFIG_MMC_SUPPORTS_TUNING=y
 CONFIG_DM_MMC=y
-CONFIG_DM_REGULATOR=y
-CONFIG_DM_REGULATOR_FIXED=y
-CONFIG_DM_REGULATOR_GPIO=y
+CONFIG_MMC_HS400_SUPPORT=y
+CONFIG_MMC_OCTEONTX=y
 CONFIG_MTD=y
 CONFIG_DM_SPI_FLASH=y
 CONFIG_SPI_FLASH=y
@@ -76,6 +76,9 @@ CONFIG_PCI=y
 CONFIG_DM_PCI=y
 CONFIG_DM_PCI_COMPAT=y
 CONFIG_PCI_OCTEONTX_ECAM=y
+CONFIG_DM_REGULATOR=y
+CONFIG_DM_REGULATOR_FIXED=y
+CONFIG_DM_REGULATOR_GPIO=y
 CONFIG_DM_SERIAL=y
 CONFIG_DEBUG_UART_PL011=y
 CONFIG_DEBUG_UART_BASE=0x87e028000000
@@ -86,9 +89,3 @@ CONFIG_DM_SPI=y
 CONFIG_OCTEONTX_SPI=y
 CONFIG_FAT_WRITE=y
 CONFIG_ERRNO_STR=y
-CONFIG_CMD_SHA1SUM=y
-CONFIG_SHA1SUM_VERIFY=y
-CONFIG_CMD_MD5SUM=y
-CONFIG_MD5SUM_VERIFY=y
-CONFIG_CMD_TIME=y
-CONFIG_CMD_FAILSAFE=y
diff --git a/configs/octeontx_81xx_defconfig b/configs/octeontx_81xx_defconfig
index 6dc69e239b..b8f3285521 100644
--- a/configs/octeontx_81xx_defconfig
+++ b/configs/octeontx_81xx_defconfig
@@ -18,9 +18,11 @@ CONFIG_HUSH_PARSER=y
 CONFIG_SYS_PROMPT="Marvell> "
 # CONFIG_CMD_BOOTEFI_HELLO_COMPILE is not set
 CONFIG_CMD_MD5SUM=y
+CONFIG_MD5SUM_VERIFY=y
 CONFIG_CMD_MEMTEST=y
 CONFIG_CMD_MX_CYCLIC=y
 CONFIG_CMD_SHA1SUM=y
+CONFIG_SHA1SUM_VERIFY=y
 # CONFIG_CMD_FLASH is not set
 CONFIG_CMD_GPIO=y
 CONFIG_CMD_I2C=y
@@ -41,6 +43,7 @@ CONFIG_CMD_SNTP=y
 CONFIG_CMD_DNS=y
 CONFIG_CMD_LINK_LOCAL=y
 CONFIG_CMD_BOOTIMGUP=y
+CONFIG_CMD_TIME=y
 CONFIG_CMD_EXT2=y
 CONFIG_CMD_EXT4=y
 CONFIG_CMD_EXT4_WRITE=y
@@ -59,15 +62,8 @@ CONFIG_DM_GPIO=y
 CONFIG_I2C_GPIO=y
 CONFIG_DM_I2C=y
 CONFIG_MISC=y
-CONFIG_MMC=y
 CONFIG_DM_MMC=y
-CONFIG_DM_REGULATOR=y
-CONFIG_DM_REGULATOR_FIXED=y
-CONFIG_DM_REGULATOR_GPIO=y
-# CONFIG_MMC_HS200_SUPPORT is not set
-# CONFIG_MMC_HS400_SUPPORT is not set
-CONFIG_MMC_WRITE=y
-# CONFIG_MMC_SUPPORTS_TUNING is not set
+CONFIG_MMC_OCTEONTX=y
 CONFIG_MTD=y
 CONFIG_MTD_PARTITIONS=y
 CONFIG_CMD_MTDPARTS=y
@@ -100,6 +96,9 @@ CONFIG_PCI=y
 CONFIG_DM_PCI=y
 CONFIG_DM_PCI_COMPAT=y
 CONFIG_PCI_OCTEONTX_ECAM=y
+CONFIG_DM_REGULATOR=y
+CONFIG_DM_REGULATOR_FIXED=y
+CONFIG_DM_REGULATOR_GPIO=y
 CONFIG_DM_RTC=y
 CONFIG_RTC_DS1307=y
 CONFIG_RTC_DS1337=y
@@ -124,8 +123,3 @@ CONFIG_USB_ETHER_ASIX88179=y
 CONFIG_USB_ETHER_RTL8152=y
 CONFIG_FAT_WRITE=y
 CONFIG_ERRNO_STR=y
-CONFIG_CMD_SHA1SUM=y
-CONFIG_SHA1SUM_VERIFY=y
-CONFIG_CMD_MD5SUM=y
-CONFIG_MD5SUM_VERIFY=y
-CONFIG_CMD_TIME=y
diff --git a/configs/octeontx_83xx_defconfig b/configs/octeontx_83xx_defconfig
index d7256b6dc0..089287f40d 100644
--- a/configs/octeontx_83xx_defconfig
+++ b/configs/octeontx_83xx_defconfig
@@ -18,9 +18,11 @@ CONFIG_HUSH_PARSER=y
 CONFIG_SYS_PROMPT="Marvell> "
 # CONFIG_CMD_BOOTEFI_HELLO_COMPILE is not set
 CONFIG_CMD_MD5SUM=y
+CONFIG_MD5SUM_VERIFY=y
 CONFIG_CMD_MEMTEST=y
 CONFIG_CMD_MX_CYCLIC=y
 CONFIG_CMD_SHA1SUM=y
+CONFIG_SHA1SUM_VERIFY=y
 # CONFIG_CMD_FLASH is not set
 CONFIG_CMD_GPIO=y
 CONFIG_CMD_I2C=y
@@ -59,15 +61,8 @@ CONFIG_DM_GPIO=y
 CONFIG_I2C_GPIO=y
 CONFIG_DM_I2C=y
 CONFIG_MISC=y
-CONFIG_MMC=y
 CONFIG_DM_MMC=y
-CONFIG_DM_REGULATOR=y
-CONFIG_DM_REGULATOR_FIXED=y
-CONFIG_DM_REGULATOR_GPIO=y
-# CONFIG_MMC_HS200_SUPPORT is not set
-# CONFIG_MMC_HS400_SUPPORT is not set
-CONFIG_MMC_WRITE=y
-# CONFIG_MMC_SUPPORTS_TUNING is not set
+CONFIG_MMC_OCTEONTX=y
 CONFIG_MTD=y
 CONFIG_DM_SPI_FLASH=y
 CONFIG_SPI_FLASH=y
@@ -95,6 +90,9 @@ CONFIG_PCI=y
 CONFIG_DM_PCI=y
 CONFIG_DM_PCI_COMPAT=y
 CONFIG_PCI_OCTEONTX_ECAM=y
+CONFIG_DM_REGULATOR=y
+CONFIG_DM_REGULATOR_FIXED=y
+CONFIG_DM_REGULATOR_GPIO=y
 CONFIG_DM_RTC=y
 CONFIG_RTC_DS1337=y
 CONFIG_SCSI=y
@@ -118,7 +116,3 @@ CONFIG_USB_ETHER_ASIX88179=y
 CONFIG_USB_ETHER_RTL8152=y
 CONFIG_FAT_WRITE=y
 CONFIG_ERRNO_STR=y
-CONFIG_CMD_SHA1SUM=y
-CONFIG_SHA1SUM_VERIFY=y
-CONFIG_CMD_MD5SUM=y
-CONFIG_MD5SUM_VERIFY=y
diff --git a/drivers/mmc/Kconfig b/drivers/mmc/Kconfig
index 7fb0a5ff25..655671cee3 100644
--- a/drivers/mmc/Kconfig
+++ b/drivers/mmc/Kconfig
@@ -38,30 +38,6 @@ config SPL_DM_MMC
 	  appear as block devices in U-Boot and can support filesystems such
 	  as EXT4 and FAT.
 
-config MMC_OCTEONTX
-	bool "Marvell OcteonTX Multimedia Card Interface support"
-	depends on (ARCH_OCTEONTX || ARCH_OCTEONTX2)
-	depends on DM_MMC
-	help
-	  This selects the OcteonTX Multimedia card Interface.
-	  If you have an OcteonTX board with a Multimedia Card slot,
-	  say Y here.  If unsure, say N.
-
-config MMC_SUPPORTS_TUNING
-	depends on MMC_OCTEONTX
-	bool "Support for HW partitioning command(eMMC)"
-	default	y
-	help
-	  This adds support for tuning in OcteonTX MMC driver.
-
-
-config MMC_HW_PARTITIONING
-	bool "Support for HW partitioning command(eMMC)"
-	default y
-	help
-	  This adds a command and an API to do hardware partitioning on eMMC
-	  devices.
-
 if MMC
 
 config MMC_SPI
@@ -307,6 +283,17 @@ config MMC_PCI
 
 	  If unsure, say N.
 
+config MMC_OCTEONTX
+	bool "Marvell OcteonTX Multimedia Card Interface support"
+	depends on (ARCH_OCTEONTX || ARCH_OCTEONTX2)
+	depends on DM_MMC
+	help
+	  This selects the OcteonTX Multimedia card Interface.
+	  If you have an OcteonTX board with a Multimedia Card slot,
+	  say Y here.
+
+	  If unsure, say N.
+
 config MMC_OMAP_HS
 	bool "TI OMAP High Speed Multimedia Card Interface support"
 	select DM_REGULATOR_PBIAS if DM_MMC && DM_REGULATOR
diff --git a/drivers/mmc/octeontx_hsmmc.c b/drivers/mmc/octeontx_hsmmc.c
index 01596334c2..7fd5c130c0 100644
--- a/drivers/mmc/octeontx_hsmmc.c
+++ b/drivers/mmc/octeontx_hsmmc.c
@@ -65,7 +65,7 @@
 
 static void octeontx_mmc_switch_to(struct mmc *mmc);
 static int octeontx_mmc_configure_delay(struct mmc *mmc);
-#if defined(CONFIG_MMC_SUPPORTS_TUNING) || defined(MMC_SUPPORTS_TUNING)
+#ifdef MMC_SUPPORTS_TUNING
 static void octeontx_mmc_set_timing(struct mmc *mmc);
 #endif
 static void set_wdog(struct mmc *mmc, u64 us);
@@ -760,7 +760,7 @@ octeontx_mmc_get_cr_mods(struct mmc *mmc, const struct mmc_cmd *cmd,
 	u8 desired_ctype = 0;
 
 	if (IS_MMC(mmc)) {
-#if defined(CONFIG_MMC_SUPPORTS_TUNING) || defined(MMC_SUPPORTS_TUNING)
+#ifdef MMC_SUPPORTS_TUNING
 		if (cmd->cmdidx == MMC_CMD_SEND_TUNING_BLOCK_HS200) {
 			if (cmd->resp_type == MMC_RSP_R1)
 				cr.rtype_xor = 1;
@@ -1569,7 +1569,7 @@ static int octeontx_mmc_dev_send_cmd(struct udevice *dev, struct mmc_cmd *cmd,
 	return octeontx_mmc_send_cmd(dev_to_mmc(dev), cmd, data);
 }
 
-#if defined(CONFIG_MMC_SUPPORTS_TUNING) || defined(MMC_SUPPORTS_TUNING)
+#ifdef MMC_SUPPORTS_TUNING
 static int octeontx_mmc_test_cmd(struct mmc *mmc, u32 opcode, int *statp)
 {
 	struct mmc_cmd cmd;
@@ -1915,7 +1915,7 @@ static int octeontx_mmc_execute_tuning(struct udevice *dev, u32 opcode)
 
 	return 0;
 }
-#endif /* defined(CONFIG_MMC_SUPPORTS_TUNING) || defined(MMC_SUPPORTS_TUNING) */
+#endif /* MMC_SUPPORTS_TUNING) */
 
 /**
  * Calculate the clock period with rounding up
@@ -2085,7 +2085,7 @@ static int octeontx_mmc_get_wp(struct udevice *dev)
 	return val;
 }
 
-#if defined(CONFIG_MMC_SUPPORTS_TUNING) || defined(MMC_SUPPORTS_TUNING)
+#ifdef MMC_SUPPORTS_TUNING
 static void octeontx_mmc_set_timing(struct mmc *mmc)
 {
 	union mio_emm_timing timing;
@@ -3051,7 +3051,7 @@ static const struct dm_mmc_ops octeontx_hsmmc_ops = {
 	.set_ios = octeontx_mmc_set_ios,
 	.get_cd = octeontx_mmc_get_cd,
 	.get_wp = octeontx_mmc_get_wp,
-#if defined(CONFIG_MMC_SUPPORTS_TUNING) || defined(MMC_SUPPORTS_TUNING)
+#ifdef MMC_SUPPORTS_TUNING
 	.execute_tuning = octeontx_mmc_execute_tuning,
 #endif
 };
diff --git a/drivers/net/octeontx/nic_main.c b/drivers/net/octeontx/nic_main.c
index e850bd6e1a..7934f9c95f 100644
--- a/drivers/net/octeontx/nic_main.c
+++ b/drivers/net/octeontx/nic_main.c
@@ -16,6 +16,7 @@
 #include <pci.h>
 #include <asm/io.h>
 
+#include <asm/arch/octeontx.h>
 #include <asm/arch/octeontx_vnic.h>
 
 #include "nic_reg.h"
@@ -437,7 +438,7 @@ static void nic_get_hw_info(struct nicpf *nic)
 
 	switch (sdevid) {
 	case PCI_SUBSYS_DEVID_88XX_NIC_PF:
-		hw->bgx_cnt = CONFIG_MAX_BGX_PER_NODE;
+		hw->bgx_cnt = MAX_BGX_PER_NODE;
 		hw->chans_per_lmac = 16;
 		hw->chans_per_bgx = 128;
 		hw->cpi_cnt = 2048;
@@ -450,7 +451,7 @@ static void nic_get_hw_info(struct nicpf *nic)
 		hw->model_id = 0x88;
 		break;
 	case PCI_SUBSYS_DEVID_81XX_NIC_PF:
-		hw->bgx_cnt = CONFIG_MAX_BGX_PER_NODE;
+		hw->bgx_cnt = MAX_BGX_PER_NODE;
 		hw->chans_per_lmac = 8;
 		hw->chans_per_bgx = 32;
 		hw->chans_per_rgx = 8;
@@ -465,7 +466,7 @@ static void nic_get_hw_info(struct nicpf *nic)
 		hw->model_id = 0x81;
 		break;
 	case PCI_SUBSYS_DEVID_83XX_NIC_PF:
-		hw->bgx_cnt = CONFIG_MAX_BGX_PER_NODE;
+		hw->bgx_cnt = MAX_BGX_PER_NODE;
 		hw->chans_per_lmac = 8;
 		hw->chans_per_bgx = 32;
 		hw->chans_per_lbk = 64;
diff --git a/drivers/net/octeontx/octeontx_bgx.c b/drivers/net/octeontx/octeontx_bgx.c
index 1993dcb4ea..952457fa51 100644
--- a/drivers/net/octeontx/octeontx_bgx.c
+++ b/drivers/net/octeontx/octeontx_bgx.c
@@ -69,9 +69,9 @@ struct bgx {
 	bool			is_rgx;
 };
 
-struct bgx_board_info bgx_board_info[CONFIG_MAX_BGX];
+struct bgx_board_info bgx_board_info[MAX_BGX_PER_NODE];
 
-struct bgx *bgx_vnic[CONFIG_MAX_BGX];
+struct bgx *bgx_vnic[MAX_BGX_PER_NODE];
 bool is_altpkg;
 extern int __cavm_if_phy_xs_init(struct mii_dev *bus, int phy_addr);
 
@@ -150,7 +150,7 @@ static bool is_bgx_port_valid(int bgx, int lmac)
 
 struct lmac *bgx_get_lmac(int node, int bgx_idx, int lmacid)
 {
-	struct bgx *bgx = bgx_vnic[(node * CONFIG_MAX_BGX_PER_NODE) + bgx_idx];
+	struct bgx *bgx = bgx_vnic[(node * MAX_BGX_PER_NODE) + bgx_idx];
 
 	if (bgx)
 		return &bgx->lmac[lmacid];
@@ -160,7 +160,7 @@ struct lmac *bgx_get_lmac(int node, int bgx_idx, int lmacid)
 
 const u8 *bgx_get_lmac_mac(int node, int bgx_idx, int lmacid)
 {
-	struct bgx *bgx = bgx_vnic[(node * CONFIG_MAX_BGX_PER_NODE) + bgx_idx];
+	struct bgx *bgx = bgx_vnic[(node * MAX_BGX_PER_NODE) + bgx_idx];
 
 	if (bgx)
 		return bgx->lmac[lmacid].mac;
@@ -170,7 +170,7 @@ const u8 *bgx_get_lmac_mac(int node, int bgx_idx, int lmacid)
 
 void bgx_set_lmac_mac(int node, int bgx_idx, int lmacid, const u8 *mac)
 {
-	struct bgx *bgx = bgx_vnic[(node * CONFIG_MAX_BGX_PER_NODE) + bgx_idx];
+	struct bgx *bgx = bgx_vnic[(node * MAX_BGX_PER_NODE) + bgx_idx];
 
 	if (!bgx)
 		return;
@@ -185,9 +185,9 @@ void bgx_get_count(int node, int *bgx_count)
 	struct bgx *bgx;
 
 	*bgx_count = 0;
-	for (i = 0; i < CONFIG_MAX_BGX_PER_NODE; i++) {
-		bgx = bgx_vnic[node * CONFIG_MAX_BGX_PER_NODE + i];
-		debug("bgx_vnic[%u]: %p\n", node * CONFIG_MAX_BGX_PER_NODE + i,
+	for (i = 0; i < MAX_BGX_PER_NODE; i++) {
+		bgx = bgx_vnic[node * MAX_BGX_PER_NODE + i];
+		debug("bgx_vnic[%u]: %p\n", node * MAX_BGX_PER_NODE + i,
 		      bgx);
 		if (bgx)
 			*bgx_count |= (1 << i);
@@ -199,7 +199,7 @@ int bgx_get_lmac_count(int node, int bgx_idx)
 {
 	struct bgx *bgx;
 
-	bgx = bgx_vnic[(node * CONFIG_MAX_BGX_PER_NODE) + bgx_idx];
+	bgx = bgx_vnic[(node * MAX_BGX_PER_NODE) + bgx_idx];
 	if (bgx)
 		return bgx->lmac_count;
 
@@ -208,7 +208,7 @@ int bgx_get_lmac_count(int node, int bgx_idx)
 
 void bgx_lmac_rx_tx_enable(int node, int bgx_idx, int lmacid, bool enable)
 {
-	struct bgx *bgx = bgx_vnic[(node * CONFIG_MAX_BGX_PER_NODE) + bgx_idx];
+	struct bgx *bgx = bgx_vnic[(node * MAX_BGX_PER_NODE) + bgx_idx];
 	u64 cfg;
 
 	if (!bgx)
@@ -245,7 +245,7 @@ void bgx_lmac_internal_loopback(int node, int bgx_idx,
 	struct lmac *lmac;
 	u64    cfg;
 
-	bgx = bgx_vnic[(node * CONFIG_MAX_BGX_PER_NODE) + bgx_idx];
+	bgx = bgx_vnic[(node * MAX_BGX_PER_NODE) + bgx_idx];
 	if (!bgx)
 		return;
 
@@ -915,7 +915,7 @@ int bgx_poll_for_link(int node, int bgx_idx, int lmacid)
 			       __func__, lmac->phydev->dev->name);
 		}
 
-#ifdef CONFIG_OCTEONTX_XCV
+#ifdef OCTEONTX_XCV
 		if (lmac->qlm_mode == QLM_MODE_RGMII)
 			xcv_setup_link(lmac->phydev->link, lmac->phydev->speed);
 #endif
@@ -1451,7 +1451,7 @@ int octeontx_bgx_probe(struct udevice *dev)
 	}
 	is_altpkg = g_cavm_bdt.alt_pkg;
 
-#ifdef CONFIG_OCTEONTX_XCV
+#ifdef OCTEONTX_XCV
 	/* Use FAKE BGX2 for RGX interface */
 	if ((((uintptr_t)bgx->reg_base >> 24) & 0xf) == 0x8) {
 		bgx->bgx_id = 2;
@@ -1471,7 +1471,7 @@ int octeontx_bgx_probe(struct udevice *dev)
 
 	node = node_id(bgx->reg_base);
 	bgx_idx = ((uintptr_t)bgx->reg_base >> 24) & 3;
-	bgx->bgx_id = (node * CONFIG_MAX_BGX_PER_NODE) + bgx_idx;
+	bgx->bgx_id = (node * MAX_BGX_PER_NODE) + bgx_idx;
 
 	if (is_board_model(CN81XX))
 		inc = 2;
@@ -1504,7 +1504,7 @@ int octeontx_bgx_probe(struct udevice *dev)
 		bgx->lmac[lmac].lmacid = lmac;
 	}
 
-#ifdef CONFIG_OCTEONTX_XCV
+#ifdef OCTEONTX_XCV
 skip_qlm_config:
 #endif
 	bgx_vnic[bgx->bgx_id] = bgx;
diff --git a/drivers/net/octeontx/octeontx_bgx.h b/drivers/net/octeontx/octeontx_bgx.h
index a00fa7a794..baf149b3e0 100644
--- a/drivers/net/octeontx/octeontx_bgx.h
+++ b/drivers/net/octeontx/octeontx_bgx.h
@@ -8,6 +8,8 @@
 #ifndef OCTEONT_BGX_H
 #define OCTEONT_BGX_H
 
+#include <asm/arch/octeontx.h>
+
 #define    MAX_LMAC_PER_BGX			4
 #define    MAX_BGX_CHANS_PER_LMAC		16
 #define    MAX_DMAC_PER_LMAC			8
@@ -15,7 +17,7 @@
 
 #define    MAX_DMAC_PER_LMAC_TNS_BYPASS_MODE	2
 
-#define    MAX_LMAC	(CONFIG_MAX_BGX_PER_NODE * MAX_LMAC_PER_BGX)
+#define    MAX_LMAC	(MAX_BGX_PER_NODE * MAX_LMAC_PER_BGX)
 
 #define    NODE_ID_MASK				0x300000000000
 #define    NODE_ID(x)				(((x) & NODE_ID_MASK) >> 44)
diff --git a/include/configs/octeontx2_95xx.h b/include/configs/octeontx2_95xx.h
index 5bdb6a1724..b94d018f59 100644
--- a/include/configs/octeontx2_95xx.h
+++ b/include/configs/octeontx2_95xx.h
@@ -1,18 +1,12 @@
-/*
+/* SPDX-License-Identifier:    GPL-2.0
  * Copyright (C) 2018 Marvell International Ltd.
  *
- * SPDX-License-Identifier:    GPL-2.0
  * https://spdx.org/licenses
  */
 
-
 #ifndef __OCTEONTX2_95XX_H__
 #define __OCTEONTX2_95XX_H__
 
-/* Generic Interrupt Controller Definitions */
-#define GICD_BASE			(0x801000000000)
-#define GICR_BASE			(0x801000002000)
-
 /* Generic Timer Definitions */
 #define COUNTER_FREQUENCY		(0x1800000)	/* 24MHz */
 
@@ -45,11 +39,8 @@
 #define CONFIG_LAST_STAGE_INIT
 /* #define CONFIG_CMD_MDIO_DBG */
 
-/**
- * Only allow the Ethernet MAC address environment variable to be
- * overwritten once.
- */
-#define CONFIG_OVERWRITE_ETHADDR_ONCE
+/* Allow environment variable to be overwritten */
+#define CONFIG_ENV_OVERWRITE
 
 /** Reduce hashes printed out */
 #define CONFIG_TFTP_TSIZE
@@ -81,7 +72,7 @@
 #if defined(CONFIG_ENV_IS_IN_MMC)
 #define CONFIG_SYS_MMC_ENV_DEV		0
 #elif defined(CONFIG_ENV_IS_IN_SPI_FLASH)
-#define CONFIG_ENV_SECT_SIZE		(64*1024)
+#define CONFIG_ENV_SECT_SIZE		(64 * 1024)
 #define CONFIG_ENV_SPI_MAX_HZ		12500000
 #define CONFIG_ENV_SPI_MODE		0
 #define CONFIG_ENV_SPI_BUS		0
@@ -103,9 +94,8 @@
 #define CONFIG_SUPPORT_EMMC_RPMB
 #define CONFIG_SUPPORT_EMMC_BOOT
 
-#define CONFIG_MMC_OCTEONTX
-#ifndef CONFIG_BLK
-# define CONFIG_BLK
+#if defined(CONFIG_MMC_OCTEONTX)
+#define MMC_SUPPORTS_TUNING
 #endif
 
 #endif /* __OCTEONTX2_95XX_H__ */
diff --git a/include/configs/octeontx2_96xx.h b/include/configs/octeontx2_96xx.h
index 79a0b2febe..0103818805 100644
--- a/include/configs/octeontx2_96xx.h
+++ b/include/configs/octeontx2_96xx.h
@@ -1,18 +1,13 @@
-/*
+/* SPDX-License-Identifier:    GPL-2.0
+ *
  * Copyright (C) 2018 Marvell International Ltd.
  *
- * SPDX-License-Identifier:    GPL-2.0
  * https://spdx.org/licenses
  */
 
-
 #ifndef __OCTEONTX2_96XX_H__
 #define __OCTEONTX2_96XX_H__
 
-/* Generic Interrupt Controller Definitions */
-#define GICD_BASE			(0x801000000000)
-#define GICR_BASE			(0x801000002000)
-
 /* Generic Timer Definitions */
 #define COUNTER_FREQUENCY		(0x1800000)	/* 24MHz */
 
@@ -45,11 +40,8 @@
 #define CONFIG_LAST_STAGE_INIT
 /* #define CONFIG_CMD_MDIO_DBG */
 
-/**
- * Only allow the Ethernet MAC address environment variable to be
- * overwritten once.
- */
-#define CONFIG_OVERWRITE_ETHADDR_ONCE
+/* Allow environment variable to be overwritten */
+#define CONFIG_ENV_OVERWRITE
 
 /** Reduce hashes printed out */
 #define CONFIG_TFTP_TSIZE
@@ -92,7 +84,7 @@
 #if defined(CONFIG_ENV_IS_IN_MMC)
 #define CONFIG_SYS_MMC_ENV_DEV		0
 #elif defined(CONFIG_ENV_IS_IN_SPI_FLASH)
-#define CONFIG_ENV_SECT_SIZE		(64*1024)
+#define CONFIG_ENV_SECT_SIZE		(64 * 1024)
 #define CONFIG_ENV_SPI_MAX_HZ		12500000
 #define CONFIG_ENV_SPI_MODE		0
 #define CONFIG_ENV_SPI_BUS		0
@@ -115,9 +107,8 @@
 #define CONFIG_SUPPORT_EMMC_RPMB
 #define CONFIG_CMD_BKOPS_ENABLE
 
-#define CONFIG_MMC_OCTEONTX
-#ifndef CONFIG_BLK
-# define CONFIG_BLK
+#if defined(CONFIG_MMC_OCTEONTX)
+#define MMC_SUPPORTS_TUNING
 #endif
 
 #endif /* __OCTEONTX2_96XX_H__ */
diff --git a/include/configs/octeontx2_98xx.h b/include/configs/octeontx2_98xx.h
index 5f57ef2847..83baa6f5c0 100644
--- a/include/configs/octeontx2_98xx.h
+++ b/include/configs/octeontx2_98xx.h
@@ -1,18 +1,13 @@
-/*
+/* SPDX-License-Identifier:    GPL-2.0
+ *
  * Copyright (C) 2018 Marvell International Ltd.
  *
- * SPDX-License-Identifier:    GPL-2.0
  * https://spdx.org/licenses
  */
 
-
 #ifndef __OCTEONTX2_98XX_H__
 #define __OCTEONTX2_98XX_H__
 
-/* Generic Interrupt Controller Definitions */
-#define GICD_BASE			(0x801000000000)
-#define GICR_BASE			(0x801000002000)
-
 /* Generic Timer Definitions */
 #define COUNTER_FREQUENCY		(0x1800000)	/* 24MHz */
 
@@ -44,11 +39,8 @@
 
 /*#define CONFIG_LAST_STAGE_INIT */
 
-/**
- * Only allow the Ethernet MAC address environment variable to be
- * overwritten once.
- */
-#define CONFIG_OVERWRITE_ETHADDR_ONCE
+/* Allow environment variable to be overwritten */
+#define CONFIG_ENV_OVERWRITE
 
 /** Reduce hashes printed out */
 #define CONFIG_TFTP_TSIZE
@@ -91,7 +83,7 @@
 #if defined(CONFIG_ENV_IS_IN_MMC)
 #define CONFIG_SYS_MMC_ENV_DEV		0
 #elif defined(CONFIG_ENV_IS_IN_SPI_FLASH)
-#define CONFIG_ENV_SECT_SIZE		(64*1024)
+#define CONFIG_ENV_SECT_SIZE		(64 * 1024)
 #define CONFIG_ENV_SPI_MAX_HZ		12500000
 #define CONFIG_ENV_SPI_MODE		0
 #define CONFIG_ENV_SPI_BUS		0
@@ -114,9 +106,8 @@
 #define CONFIG_SUPPORT_EMMC_RPMB
 #define CONFIG_CMD_BKOPS_ENABLE
 
-#define CONFIG_MMC_OCTEONTX
-#ifndef CONFIG_BLK
-# define CONFIG_BLK
+#if defined(CONFIG_MMC_OCTEONTX)
+#define MMC_SUPPORTS_TUNING
 #endif
 
 #endif /* __OCTEONTX2_98XX_H__ */
diff --git a/include/configs/octeontx2_loki.h b/include/configs/octeontx2_loki.h
index dc4f62b228..d39eeb7383 100644
--- a/include/configs/octeontx2_loki.h
+++ b/include/configs/octeontx2_loki.h
@@ -1,17 +1,13 @@
-/*
+/* SPDX-License-Identifier:    GPL-2.0
+ *
  * Copyright (C) 2019 Marvell International Ltd.
  *
- * SPDX-License-Identifier:    GPL-2.0
  * https://spdx.org/licenses
  */
 
 #ifndef __OCTEONTX2_LOKI_H__
 #define __OCTEONTX2_LOKI_H__
 
-/* Generic Interrupt Controller Definitions */
-#define GICD_BASE			(0x801000000000)
-#define GICR_BASE			(0x801000002000)
-
 /* Generic Timer Definitions */
 #define COUNTER_FREQUENCY		(0x1800000)	/* 24MHz */
 
@@ -44,11 +40,8 @@
 #define CONFIG_LAST_STAGE_INIT
 /* #define CONFIG_CMD_MDIO_DBG */
 
-/**
- * Only allow the Ethernet MAC address environment variable to be
- * overwritten once.
- */
-#define CONFIG_OVERWRITE_ETHADDR_ONCE
+/* Allow environment variable to be overwritten */
+#define CONFIG_ENV_OVERWRITE
 
 /** Reduce hashes printed out */
 #define CONFIG_TFTP_TSIZE
@@ -102,9 +95,8 @@
 #define CONFIG_SUPPORT_EMMC_RPMB
 #define CONFIG_SUPPORT_EMMC_BOOT
 
-#define CONFIG_MMC_OCTEONTX
-#ifndef CONFIG_BLK
-# define CONFIG_BLK
+#if defined(CONFIG_MMC_OCTEONTX)
+#define MMC_SUPPORTS_TUNING
 #endif
 
 #endif /* __OCTEONTX2_LOKI_H__ */
diff --git a/include/configs/octeontx_83xx.h b/include/configs/octeontx_83xx.h
deleted file mode 100644
index 691448abc8..0000000000
--- a/include/configs/octeontx_83xx.h
+++ /dev/null
@@ -1,127 +0,0 @@
-/*
- * Copyright (C) 2018 Marvell International Ltd.
- *
- * SPDX-License-Identifier:    GPL-2.0
- * https://spdx.org/licenses
- */
-
-
-#ifndef __OCTEONTX_83XX_H__
-#define __OCTEONTX_83XX_H__
-
-/* Generic Interrupt Controller Definitions */
-#define GICD_BASE			(0x801000000000)
-#define GICR_BASE			(0x801000002000)
-
-/* Generic Timer Definitions */
-#define COUNTER_FREQUENCY		(0x1800000)	/* 24MHz */
-
-#define CONFIG_SUPPORT_RAW_INITRD
-
-/** Maximum size of image supported for bootm (and bootable FIT images) */
-#define CONFIG_SYS_BOOTM_LEN		(256 << 20)
-
-/** Memory base address */
-#define CONFIG_SYS_SDRAM_BASE		CONFIG_SYS_TEXT_BASE
-
-/** Stack starting address */
-#define CONFIG_SYS_INIT_SP_ADDR		(CONFIG_SYS_SDRAM_BASE + 0xffff0)
-
-/** Memory test starting address */
-#define CONFIG_SYS_MEMTEST_START	CONFIG_SYS_SDRAM_BASE
-
-/** Memory test end address */
-#define CONFIG_SYS_MEMTEST_END		(CONFIG_SYS_SDRAM_BASE + 0xf0000)
-
-/** Heap size for U-Boot */
-#define CONFIG_SYS_MALLOC_LEN		(CONFIG_ENV_SIZE + 64 * 1024 * 1024)
-
-#define CONFIG_SYS_LOAD_ADDR		CONFIG_SYS_SDRAM_BASE
-
-#define CONFIG_NR_DRAM_BANKS		1
-
-/**
- * Only allow the Ethernet MAC address environment variable to be
- * overwritten once.
- */
-#define CONFIG_OVERWRITE_ETHADDR_ONCE
-
-/** Maximum number of BGX interfaces per CPU node */
-#define CONFIG_MAX_BGX_PER_NODE		4
-
-/** Maximum total number of BGX interfaces across all nodes */
-#define CONFIG_MAX_BGX			4
-
-#define CONFIG_MISC_INIT_R
-
-/** Reduce hashes printed out */
-#define CONFIG_TFTP_TSIZE
-
-/* BOOTP options */
-#define CONFIG_BOOTP_BOOTFILESIZE
-#define CONFIG_BOOTP_BOOTPATH
-#define CONFIG_BOOTP_GATEWAY
-#define CONFIG_BOOTP_HOSTNAME
-#define CONFIG_BOOTP_PXE
-#define CONFIG_BOOTP_TFTP_SERVERIP
-
-/* AHCI support Definitions */
-#ifdef CONFIG_DM_SCSI
-/** Maximum number of SATA devices per controller*/
-#define CONFIG_SYS_SCSI_MAX_SCSI_ID	1
-/** Enable 48-bit SATA addressing */
-# define CONFIG_LBA48
-/** Enable 64-bit addressing */
-# define CONFIG_SYS_64BIT_LBA
-#endif
-
-/***** SPI Defines *********/
-#ifdef CONFIG_DM_SPI_FLASH
-#define CONFIG_SF_DEFAULT_SPEED 12500000
-#define CONFIG_SF_DEFAULT_MODE	0
-#define CONFIG_SF_DEFAULT_BUS	0
-#define CONFIG_SF_DEFAULT_CS	0
-#endif
-
-/** Extra environment settings */
-#define CONFIG_EXTRA_ENV_SETTINGS	\
-					"loadaddr=040080000\0"	\
-					"autoload=0\0"
-
-/** Environment defines */
-#define CONFIG_ENV_SIZE			0x8000
-#define CONFIG_ENV_OFFSET		0xf00000
-#if defined(CONFIG_ENV_IS_IN_MMC)
-#define CONFIG_SYS_MMC_ENV_DEV		0
-#elif defined(CONFIG_ENV_IS_IN_SPI_FLASH)
-#define CONFIG_ENV_SECT_SIZE		(64*1024)
-#define CONFIG_ENV_SPI_MAX_HZ		12500000
-#define CONFIG_ENV_SPI_MODE		0
-#define CONFIG_ENV_SPI_BUS		0
-#define CONFIG_ENV_SPI_CS		0
-#endif
-
-/* Monitor Command Prompt */
-#define CONFIG_SYS_CBSIZE		1024	/** Console I/O Buffer Size */
-#define CONFIG_SYS_BARGSIZE		CONFIG_SYS_CBSIZE
-
-#define CONFIG_SYS_MAXARGS		64	/** max command args */
-
-#define CONFIG_SYS_MMC_MAX_BLK_COUNT	8191
-
-#undef CONFIG_SYS_PROMPT
-#define CONFIG_SYS_PROMPT		env_get("prompt")
-
-#define CONFIG_HW_WATCHDOG
-
-/** EMMC specific defines */
-#define CONFIG_SUPPORT_EMMC_BOOT
-#define CONFIG_SUPPORT_EMMC_RPMB
-#define CONFIG_CMD_BKOPS_ENABLE
-
-#define CONFIG_MMC_OCTEONTX
-#ifndef CONFIG_BLK
-# define CONFIG_BLK
-#endif
-
-#endif /* __OCTEONTX_83XX_H__ */
diff --git a/include/configs/octeontx_81xx.h b/include/configs/octeontx_common.h
similarity index 78%
rename from include/configs/octeontx_81xx.h
rename to include/configs/octeontx_common.h
index 22543e9e0c..43483feedb 100644
--- a/include/configs/octeontx_81xx.h
+++ b/include/configs/octeontx_common.h
@@ -1,17 +1,12 @@
-/*
+/* SPDX-License-Identifier:    GPL-2.0
+ *
  * Copyright (C) 2018 Marvell International Ltd.
  *
- * SPDX-License-Identifier:    GPL-2.0
  * https://spdx.org/licenses
  */
 
-
-#ifndef __OCTEONTX_81XX_H__
-#define __OCTEONTX_81XX_H__
-
-/* Generic Interrupt Controller Definitions */
-#define GICD_BASE			(0x801000000000)
-#define GICR_BASE			(0x801000002000)
+#ifndef __OCTEONTX_COMMON_H__
+#define __OCTEONTX_COMMON_H__
 
 /* Generic Timer Definitions */
 #define COUNTER_FREQUENCY		(0x1800000)	/* 24MHz */
@@ -40,18 +35,8 @@
 
 #define CONFIG_NR_DRAM_BANKS		1
 
-/**
- * Only allow the Ethernet MAC address environment variable to be
- * overwritten once.
- */
-#define CONFIG_OVERWRITE_ETHADDR_ONCE
-
-/** Maximum number of BGX interfaces per CPU node */
-#define CONFIG_MAX_BGX_PER_NODE		3
-
-/** Maximum total number of BGX interfaces across all nodes */
-#define CONFIG_MAX_BGX			3
-#define CONFIG_OCTEONTX_XCV
+/* Allow environment variable to be overwritten */
+#define CONFIG_ENV_OVERWRITE
 
 #define CONFIG_MISC_INIT_R
 
@@ -95,7 +80,7 @@
 #if defined(CONFIG_ENV_IS_IN_MMC)
 #define CONFIG_SYS_MMC_ENV_DEV		0
 #elif defined(CONFIG_ENV_IS_IN_SPI_FLASH)
-#define CONFIG_ENV_SECT_SIZE		(64*1024)
+#define CONFIG_ENV_SECT_SIZE		(64 * 1024)
 #define CONFIG_ENV_SPI_MAX_HZ		12500000
 #define CONFIG_ENV_SPI_MODE		0
 #define CONFIG_ENV_SPI_BUS		0
@@ -120,9 +105,4 @@
 #define CONFIG_SUPPORT_EMMC_RPMB
 #define CONFIG_CMD_BKOPS_ENABLE
 
-#define CONFIG_MMC_OCTEONTX
-#ifndef CONFIG_BLK
-# define CONFIG_BLK
-#endif
-
-#endif /* __OCTEONTX_81XX_H__ */
+#endif /* __OCTEONTX_COMMON_H__ */
-- 
2.29.0

